In [1]:
import numpy as np
from sklearn.model_selection import cross_val_score, KFold
from sklearn.linear_model import LinearRegression
from sklearn.datasets import load_wine

In [5]:
data = load_wine()
X, y = data.data, data.target

**data.data** - таблица признаков

**data.target** - правильные ответы

In [2]:
model = LinearRegression()

kf = KFold(n_splits=5, shuffle=True, random_state=42)

**n_splits** - количество фолдов

**shuffle** - перемешивает данные перед разбиением

**random_state** - фиксирует случайность

In [7]:
scores = cross_val_score(model, X, y, cv=kf, scoring="r2")

**scoring** - метрика качества

In [12]:
for i, score in enumerate(scores, start=1):
    print(f"Fold {i}: {score:.3f}")
print(f"Среднее: {np.mean(scores):.3f}")
print(f"Разброс: {np.std(scores):.3f}")

Fold 1: 0.883
Fold 2: 0.889
Fold 3: 0.862
Fold 4: 0.900
Fold 5: 0.851

Среднее: 0.877
Разброс: 0.018


По результату можно понять что модель объясняет ~87.7% дисперсии целевой переменной.

По фолдам:
 - Все значения между 0.851 и 0.900 - модель стабильна
 - Разброс (стандартное отклонение) 0.018 - маленький, все результаты держатся близко к среднему 0.877, отклоняясь примерно на ±0.018